pkg/proxy/proxy.go

```diff
@@ -286,12 +288,44 @@ func uploadToBlobStorage(ctx context.Context, cacheID int) (*DockerGHAUploadCach
 			}
 		}
 
+	case ProviderGCS:
+		if cacheEntry.BackendReserveResponse.GCS.ShortLivedToken == nil {
+			return nil, fmt.Errorf("no short lived token found")
+		}
+
+		creds := option.WithCredentialsJSON([]byte(cacheEntry.BackendReserveResponse.GCS.ShortLivedToken.AccessToken))
+		client, err := storage.NewClient(ctx, creds)
 		if err != nil {
-			return nil, fmt.Errorf("upload to S3 failed after %d attempts: %w", maxRetries, err)
+			return nil, fmt.Errorf("failed to create GCS client: %w", err)
 		}
 
-	} else {
+		defer client.Close()
+
+		bucket := client.Bucket(cacheEntry.BackendReserveResponse.GCS.BucketName)
+		object := bucket.Object(cacheEntry.BackendReserveResponse.GCS.CacheKey)
+
+		wc := object.NewWriter(ctx)
```

Comment:
We set content type in S3. Do we need it here? 

Low priority: we can also play with ChunkSize for different sized objects.

Reply:
Changed logic to always call Close()

Reply:
@prajjwaldimri From the docs,

> Since writes happen asynchronously, `Write` may return a nil error even though the write failed (or will fail). Always use the error returned from `Writer.Close` to determine if the upload was successful.

We should move the `Close` operation inside the attempt loop.

Reply:
Moved